toolkit/tools/pkg/imagecustomizerlib/baseconfigs.go

```diff
@@ -0,0 +1,169 @@
+package imagecustomizerlib
+
+import (
+	"context"
+	"fmt"
+
+	"path/filepath"
+
+	"github.com/microsoft/azure-linux-image-tools/toolkit/tools/imagecustomizerapi"
+)
+
+func ResolveBaseConfigs(ctx context.Context, rc *ResolvedConfig) error {
+	visited := make(map[string]bool)
+	pathStack := []string{}
+
+	configChain, err := BuildInheritanceChain(ctx, rc.Config, rc.BaseConfigPath, visited, pathStack)
+	if err != nil {
+		return err
+	}
+
+	merged := &ResolvedConfig{}
+
+	resolveOverrideFields(configChain, merged)
+	resolveMergeFields(configChain, merged)
+
+	if merged.Config.Input != (imagecustomizerapi.Input{}) {
+		rc.Config.Input = merged.Config.Input
+	}
+	if merged.Config.Output != (imagecustomizerapi.Output{}) {
+		rc.Config.Output = merged.Config.Output
+	}
+
+	return nil
+}
+
+func BuildInheritanceChain(ctx context.Context, cfg *imagecustomizerapi.Config, baseDir string, visited map[string]bool, pathStack []string) ([]*imagecustomizerapi.Config, error) {
+	if cfg.BaseConfigs != nil {
+		for i, base := range cfg.BaseConfigs {
+			err := base.IsValid()
+			if err != nil {
+				return nil, fmt.Errorf("invalid baseConfig item at index %d:\n%w", i, err)
+			}
+		}
+	}
+
+	var chain []*imagecustomizerapi.Config
+
+	for _, base := range cfg.BaseConfigs {
+		basePath := filepath.Join(baseDir, base.Path)
+		absPath, err := filepath.Abs(basePath)
```

Comment:
Use `file.GetAbsPathWithBase` instead of `filepath.Join` and `filepath.Abs`. This will properly handle the case where the base config path is specified as an absolute path.